### ISSUE TITLE
OCI Runtime error - gunicorn executable file not found in PATH

### ISSUE THREAD

**Issue by ChaiBapchya** [edgi-govdata-archiving/web-monitoring-processing#36]
# Command
`docker run -t yay`

# Output
```
docker: Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "exec: \"gunicorn\": executable file not found in $PATH".
```

**Comment by ChaiBapchya** [comment]
When I tried to look at the Dockerfile
Following two lines were commented
```
RUN apt-get install -y gunicorn
RUN apt-get install -y python-gevent
```

Looked like that was the issue. The executable file of gunicorn didn't exist because it wasn't installed in the first place.
However after docker build and subsequent docker run
Following error -
```
Traceback (most recent call last):
  File "/usr/bin/gunicorn", line 9, in <module>
    load_entry_point('gunicorn==19.0.0', 'console_scripts', 'gunicorn')()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 74, in run
    WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/base.py", line 166, in run
    super(Application, self).run()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/base.py", line 71, in run
    Arbiter(self).run()
  File "/usr/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 169, in run
    self.manage_workers()
  File "/usr/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 477, in manage_workers
    self.spawn_workers()
  File "/usr/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 537, in spawn_workers
    time.sleep(0.1 * random.random())
  File "/usr/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 209, in handle_chld
    self.reap_workers()
  File "/usr/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 459, in reap_workers
    raise HaltServer(reason, self.WORKER_BOOT_ERROR)
gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
```

**Comment by danielballan** [comment]
@lh00000000, do you have the bandwidth to take a look at this?

**Comment by lh00000000** [comment]
hi, 
sorry about that. It was a hastily pasted-from-my-snippets dockerfile. 
the reason the gunicorn line was commented out was because I think it's more reliable to use gunicorn when it's installed by pip. There were just a few work-related situations I've seen where I could not use the pip solution, thus I had kept those lines there. 

I've made a PR to fix this here: #50 although I would say that I recommend ignoring details of deploying this newspaper thing. The original intent of this experiment was more about the proposal that the concept of diffing should be treated as one-or-more parallel problems (e.g. diffing the html string, diffing the text nodes only, diffing the article-text-only output, etc). I didn't know what kind of architecture/hosting the project had, so I thought a dockerized webservice of the first html-to-plaintext module i could find would be the most portable+minimal thing to make at the time.

**Comment by danielballan** [comment]
Thanks, @lh00000000. Yeah, we may end up rethinking the deployment. We're _just_ getting to that part. But it's cool to have something working for others to tinker with, so I'll go ahead and merge #50 even if we are likely to revisit it.
